### ISSUE TITLE
Is it possible to make available packages depend on system?

### ISSUE THREAD

**Issue by rickynils** [nix-community/flakelight#21]
Sometimes I have specific packages that I don't want to build for specific systems. Is it possible to somehow specify this in flakelight? The only way I found is this:

```
packages = {
  my-package = { stdenv, ... }: if stdenv.hostPlatform.system == "x86_64-linux" then null else ...
}
```

This makes the package disappear from `nix flake show` if I'm on an `x86_64-linux` system. However, if I'm on another system the package will actually be visible (because it is not evaluated to `null` unless `--all-systems` is used). This feels a bit hacky and inconvenient.

**Comment by accelbread** [comment]
Hmm, thats likely the best way at the moment, except for using `outputs.packages` instead of `packages`. 

I'll see about fixing this.

Probably, I'll make it so that packages gets a system arg with the system it is generating for.

**Comment by rickynils** [comment]
A system arg sounds great. Having that as a top-level arg would be awesome, but per-package works too.

For `checks` I also need to do this, but it is slightly cleaner because you can already write checks like `pkgs: <attrs of checks>`, so you don't have to evaluate `pkgs.stdenv` for each check. A system arg would feel even better though.

**Comment by rickynils** [comment]
So, setting a package to `null` doesn't really work. For example, when using `--all-systems` you will then get errors because the package checks doesn't handle the fact that a package can be null. Also devShells seem to fail evaluate, but I'm not sure why.

**Comment by accelbread** [comment]
Apologies for the delay, have been busy.

I've fixed this in 32bf894a5033451cc1bc252086e33c0e1ef6859e.

Now if `packages` is a function it also gets a `system` arg. An example of use is in the test added along with the change.

Unfortunately, unlike `checks`, `packages` can't get `pkgs` as an arg since that would cause infinite recursion (`pkgs` is defined in terms of `packages`). It can however get the system.

Also, in `checks`, you don't need to look at `pkgs.stdenv`, can just use `pkgs.system`. 

Thanks for the report!

**Comment by accelbread** [comment]
Added docs in f34d07dd57bf0df4d0695a1067b3c12307f2bd5e.
